test/test_irods_http_api.py

```diff
@@ -1648,6 +1650,177 @@ def test_parallel_writes(self):
         self.assertEqual(r.status_code, 200)
         self.assertEqual(r.json()['irods_response']['status_code'], 0)
 
+    def test_parallel_writes_with_data_exceeding_internal_write_threshold(self):
+        # This test assumes the HTTP API is configured to use a value smaller than
+        # 96kb for "/irods_client/buffer_size_in_bytes_for_write_operations". This is
```

Comment:
If it's possible for the HTTP API to check its own configuration, we can skip or fail the test if it's not meeting the assumptions. Just a thought; don't have to do that.

Reply:
The tests do not have that ability right now. We can add that though.

That information could be exposed via the /info endpoint. The question then becomes, is that information valuable to users or is it only valuable to the tests.

Please open an issue for that.

Reply:
Aren't these tests running on the same machine as the HTTP API server? If so, we could just crack open the config file, right? Sorry if these are naive questions.

Reply:
It's fine. Ask all the questions.

> Aren't these tests running on the same machine as the HTTP API server?

They aren't required to run on the same machine. The config.py provides information for helping the tests run against local or remote HTTP API servers. The tests do not assume the HTTP API is local to them.

I run the tests locally, but others have been known to run/test it remotely.

Reply:
Okay, that makes sense. Maybe we can set the configurations in question in the test config and leave a comment referencing this test should anybody ever change these? Just concerned about the fact that these tests can (and will?) pass even if the assumptions are not met.

Anyway, I created https://github.com/irods/irods_client_http_api/issues/322 for the configuration fetching mechanism.

Reply:
That's a possibility. The GenQuery2 tests are controlled by config.py just like you've described.

Reply:
Resolving now that we have issue #322.